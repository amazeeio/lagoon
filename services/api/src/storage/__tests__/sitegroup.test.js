// @flow

/* eslint-disable no-undef */

const path = require('path');
const {
  siteGroupsFilePath,
  readSiteGroupsFile,
} = require('../../storage/sitegroup');

const HIERA_REPO = path.join(__dirname, '__fixtures__', 'hiera1');

describe('siteGroupsFilePath', () => {
  test('should return a proper filepath to the sitegroup YAML', () => {
    const p = siteGroupsFilePath(HIERA_REPO);
    expect(p).toEqual(path.join(HIERA_REPO, 'amazeeio', 'sitegroups.yaml'));
  });
});

describe('readSiteGroupsFile', () => {
  test('should return the string content for api-test-hiera', async () => {
    const content = await readSiteGroupsFile(HIERA_REPO);
    expect(content).toMatchSnapshot();
  });
});
