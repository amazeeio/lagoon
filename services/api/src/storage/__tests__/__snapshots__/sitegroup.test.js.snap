// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readSiteGroupsFile should return the string content for api-test-hiera 1`] = `
Object {
  "amazeeio_sitegroups": Object {
    "ci-node1": Object {
      "active_systems": Object {
        "deploy": "lagoon_openshift",
        "remove": "lagoon_openshift",
      },
      "client": "deploytestclient",
      "git_url": "git@git:/git/node.git",
      "openshift": Object {
        "console": "https://192.168.77.100:8443/",
        "password": "developer",
        "project": "myproject",
        "registry": "docker-registry-default.192.168.77.100.nip.io:443",
        "router_pattern": "\${sitegroup}.\${branch}.192.168.77.100.nip.io",
        "template": "node.yml",
        "username": "developer",
      },
      "slack": Object {
        "channel": "amazeeio-testing",
        "webhook": "https://hooks.slack.com/services/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",
      },
    },
    "ci-node2": Object {
      "active_systems": Object {
        "deploy": "lagoon_openshift",
        "remove": "lagoon_openshift",
      },
      "client": "deploytestclient",
      "git_url": "git@git:/git/node.git",
      "openshift": Object {
        "console": "https://192.168.77.100:8443/",
        "password": "developer",
        "project": "myproject",
        "registry": "docker-registry-default.192.168.77.100.nip.io:443",
        "router_pattern": "\${sitegroup}.\${branch}.192.168.77.100.nip.io",
        "template": "node.yml",
        "username": "developer",
      },
      "slack": Object {
        "channel": "amazeeio-testing",
        "webhook": "https://hooks.slack.com/services/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",
      },
    },
    "ci-node_subfolder1": Object {
      "active_systems": Object {
        "deploy": "lagoon_openshift",
        "remove": "lagoon_openshift",
      },
      "client": "deploytestclient",
      "git_url": "git@git:/git/node_subfolder.git",
      "openshift": Object {
        "console": "https://192.168.77.100:8443/",
        "folder": "subfolder",
        "password": "developer",
        "project": "myproject",
        "registry": "docker-registry-default.192.168.77.100.nip.io:443",
        "router_pattern": "\${sitegroup}.\${branch}.192.168.77.100.nip.io",
        "template": "node.yml",
        "username": "developer",
      },
      "slack": Object {
        "channel": "amazeeio-testing",
        "webhook": "https://hooks.slack.com/services/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",
      },
    },
    "ci-node_subfolder2": Object {
      "active_systems": Object {
        "deploy": "lagoon_openshift",
        "remove": "lagoon_openshift",
      },
      "client": "deploytestclient",
      "git_url": "git@git:/git/node_subfolder.git",
      "openshift": Object {
        "console": "https://192.168.77.100:8443/",
        "folder": "subfolder",
        "password": "developer",
        "project": "myproject",
        "registry": "docker-registry-default.192.168.77.100.nip.io:443",
        "router_pattern": "\${sitegroup}.\${branch}.192.168.77.100.nip.io",
        "template": "node.yml",
        "username": "developer",
      },
      "slack": Object {
        "channel": "amazeeio-testing",
        "webhook": "https://hooks.slack.com/services/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",
      },
    },
  },
}
`;
